hey merge me, signup and login is working

Minhtri7496 opened the pull request.

Jonno1809 committed:
Merge pull request #3 from Jonno1809/firebase

Add Firebase database provider

Minhtri7496 committed:
Merge https://github.com/Jonno1809/UC-Marketplace into develop

Minhtri7496 committed:
add logindetail page

Minhtri7496 committed:
login and signup linked to database

Minhtri7496 committed:
signup linked to userdetail database

Jonno1809 commented on src/app/app.component.ts:
```diff
@@ -9,14 +9,15 @@ import { LoginPage } from '../pages/login/login';
 import { SignupPage} from '../pages/signup/signup';
 import { ItemPage } from '../pages/item/item';
 import { FirebaseTestPage } from '../pages/firebase-test/firebase-test';
+import {LogindetailPage} from '../pages/logindetail/logindetail';
```
For consistency, spaces between curly brackets would be good i.e. { Module } instead of {Module}

Jonno1809 commented on src/app/app.module.ts:
```diff
@@ -17,10 +17,11 @@ import 'firebase/storage';
 import { StatusBar } from '@ionic-native/status-bar';
 import { SplashScreen } from '@ionic-native/splash-screen';
-
+import {LogindetailPage} from '../pages/logindetail/logindetail';
```
Same here

Jonno1809 commented on src/models/user.ts:
```diff
@@ -0,0 +1,10 @@
+export interface User{
+
+email: string;
```
Indentation for needed here for readability (and consistency)

Jonno1809 replied:
Same to the ones below

Jonno1809 commented on src/pages/item/item.ts:
```diff
@@ -1,5 +1,9 @@
 import { Component } from '@angular/core';
-import { IonicPage, NavController, NavParams } from 'ionic-angular';
+import { IonicPage, NavController, NavParams, ToastController } from 'ionic-angular';
+import{AngularFireAuth} from 'angularfire2/auth';
```
Spacessssss

Jonno1809 commented on src/pages/item/item.ts:
```diff
@@ -30,7 +37,22 @@ export class ItemPage {
+        
+        duration:3000
+      }).present();
+    }else{
```
Line up that curly bracket with the if statement please

Jonno1809 replied:
Actually it's more like that if statement should be indented to match the rest of it

Jonno1809 commented on src/pages/item/item.ts:
```diff
@@ -30,7 +37,22 @@ export class ItemPage {
+    console.log(data.email);
+  }
+  );
   }
```
2 ending curly brackets in the same line? Indent please

Jonno1809 commented on src/pages/login/login.ts:
```diff
@@ -2,6 +2,7 @@ import { Component } from '@angular/core';
 import { IonicPage, NavController, NavParams } from 'ionic-angular';
 import {HomePage} from '../home/home'
 import{SignupPage} from '../signup/signup'
+import{LogindetailPage} from '../logindetail/logindetail';
```
Space

Jonno1809 commented on src/pages/login/login.ts:
```diff
@@ -24,7 +25,7 @@ export class LoginPage {
   loginTapped(event) {
     // That's right, we're pushing to ourselves!
-    this.navCtrl.push(HomePage    );}
+    this.navCtrl.push(LogindetailPage    );}
```
What on earth is that big space after LogindetailPage, fix please

Jonno1809 replied:
And that ending curly bracket? Why there? Put it a line down

Jonno1809 commented on src/pages/login/login.ts:
```diff
@@ -24,7 +25,7 @@ export class LoginPage {
+    this.navCtrl.push(LogindetailPage    );}
     signupTapped(event) {
       // That's right, we're pushing to ourselves!
       this.navCtrl.push(SignupPage    );}
```
Same deal

Jonno1809 commented on src/pages/logindetail/logindetail.html:
```diff
@@ -0,0 +1,42 @@
+          <div class="logo">
+            <img src="assets/images/marketplace icon.png" width = "200px" alt="">
+            <img src="assets/images/title.png" width = "250px" alt="">
+          </div></div>
```
Press enter between the two \</div> tags please

Jonno1809 commented on src/pages/logindetail/logindetail.ts:
```diff
@@ -0,0 +1,46 @@
+  }
+  signupTapped(event) {
+    // That's right, we're pushing to ourselves!
+    this.navCtrl.push(SignupPage    );}
```
Space and curly bracket again

Jonno1809 commented on src/pages/signup/signup.html:
```diff
@@ -5,45 +5,56 @@
+                <div class="logo">
+                  <img src="assets/images/marketplace icon.png" width = "200px" alt="">
+                  <img src="assets/images/title.png" width = "250px" alt="">
+                </div></div>
```
Separate \</div>'s

Jonno1809 commented on src/pages/signup/signup.html:
```diff
@@ -5,45 +5,56 @@
+                  <ion-item class="listcolor" >
+                    <ion-label floating>Last name </ion-label>
+                    <ion-input type="text" [(ngModel)]="user.lastname"></ion-input>
+                </ion-item>
```
This and the next ion-items are out of line

Jonno1809 commented on src/pages/logindetail/logindetail.ts:
```diff
@@ -0,0 +1,46 @@
+import {User} from '../../models/user'
+import { AngularFireAuth } from 'angularfire2/auth';
+import {ItemPage} from '../item/item';
+import {SignupPage} from '../signup/signup';
```
Inconsistencies

Jonno1809 reviewed (changes requested):
Quite a lot of code style inconsistencies, please fix to increase readability and avoid future errors (plus they make my eyes bleed :stuck_out_tongue: ).

Other than that, looks good.

Minhtri7496 committed:
Signup links to userdetail

this userdetail is saved using UID from authentication database so if you want to pull data for, try to retrieve UID from authentication database and then compare it with the UID under userdetail

Minhtri7496 committed:
signup updated

Jonno1809 commented:
Fine, I'll let those coding inconsistencies slide (just this once :wink: )
